lib/stan_math_2.6.3/stan/math/prim/scal/prob/wiener_log.hpp

```diff
@@ -0,0 +1,224 @@
+      using boost::math::tools::promote_args;
+      using boost::math::isinf;
+      using boost::math::isfinite;
+      using std::log;
+      using std::exp;
+      using std::pow;
+
+      if (!(stan::length(y)
+        && stan::length(alpha)
+        && stan::length(beta)
+        && stan::length(tau)
+        && stan::length(delta)))
+        return 0.0;
+
+      typedef typename return_type<T_y, T_alpha, T_tau,
+                                   T_beta, T_delta>::type T_return_type;
+      T_return_type lp(0.0);
+
+      check_not_nan(function, "Random variable", y);
+      check_not_nan(function, "Boundary separation", alpha);
+      check_not_nan(function, "A-priori bias", beta);
+      check_not_nan(function, "Nondecision time", tau);
+      check_not_nan(function, "Drift rate", delta);
+      check_finite(function, "Boundary separation", alpha);
+      check_finite(function, "A-priori bias", beta);
+      check_finite(function, "Nondecision time", tau);
+      check_finite(function, "Drift rate", delta);
+      check_positive(function, "Random variable", y);
+      check_positive(function, "Boundary separation", alpha);
+      check_positive(function, "Nondecision time", tau);
+      check_bounded(function, "A-priori bias", beta , 0, 1);
+      check_consistent_sizes(function,
+       "Random variable", y,
+       "Boundary separation", alpha,
+       "A-priori bias", beta,
+       "Nondecision time", tau,
+       "Drift rate", delta);
+
+      size_t N = std::max(max_size(y, alpha, beta), max_size(tau, delta));
+      if (!N)
+        return 0.0;
+      VectorView<const T_y> y_vec(y);
+      VectorView<const T_alpha> alpha_vec(alpha);
+      VectorView<const T_beta> beta_vec(beta);
+      VectorView<const T_tau> tau_vec(tau);
+      VectorView<const T_delta> delta_vec(delta);
+
+      if (!include_summand<propto, T_y, T_alpha, T_tau,
+          T_beta, T_delta>::value) {
+        return 0;
+      }
+
+      for (size_t i = 0; i < N; i++)
+        if (y_vec[i] < tau_vec[i]) {
+          lp = stan::math::negative_infinity();
+          return lp;
+        }
+
+        for (size_t i = 0; i < N; i++) {
+          typename scalar_type<T_beta>::type one_minus_beta = 1.0 - beta_vec[i];
+          typename scalar_type<T_alpha>::type alpha2 = alpha_vec[i] *
+                                                       alpha_vec[i];
+          T_return_type x = y_vec[i];
+          T_return_type kl, ks, tmp = 0;
+          T_return_type k, K;
+
+
+        x = x - tau_vec[i];  // remove non-decision time from x
+        x = x / alpha2;  // convert t to normalized time tt
+
+        // calculate number of terms needed for large t:
+        // if error threshold is set low enough
+        if (MY_PI * x * WIENER_ERR < 1) {
+          // compute bound
+          kl = sqrt(-2.0 * log(MY_PI * x * WIENER_ERR) / (pow(MY_PI, 2) * x));
+          // ensure boundary conditions met
+          kl = (kl > 1.0 / (MY_PI * sqrt(x))) ? kl : 1.0 / (MY_PI * sqrt(x));
```

Comment:
calculate `sqrt(x)` and save it for reuse.   `sqrt()` is a very expensive function!